src/check/unify.zig

```diff
@@ -1576,6 +1576,21 @@ fn Unifier(comptime StoreTypeB: type) type {
                                 .int => return error.TypeMismatch,
                             }
                         },
+                        .num_unbound => |b_num_unbound| {
+                            // Check if the compact type satisfies the requirements
+                            switch (a_num_compact) {
+                                .int => |int_prec| {
+                                    const result = self.checkIntPrecisionRequirements(int_prec, b_num_unbound);
+                                    switch (result) {
+                                        .ok => {},
+                                        .negative_unsigned => return error.NegativeUnsignedInt,
+                                        .too_large => return error.NumberDoesNotFit,
+                                    }
+                                },
+                                .frac => return error.TypeMismatch,
+                            }
+                            self.merge(vars, vars.b.desc.content);
```

Comment:
This should be `self.merge(vars, vars.a.desc.content);`, since `a` is more specific than `b` here!